src/libecalc/application/graph_result.py

```diff
@@ -130,22 +164,38 @@ def _evaluate_installations(self, variables_map: dto.VariablesMap) -> List[libec
             ]
 
             installation_node_info = self.graph.get_node_info(installation.id)
-            power = reduce(
-                operator.add,
-                [
-                    TimeSeriesRate.from_timeseries_stream_day_rate(component.power, regularity=regularity)
-                    for component in sub_components
-                    if self.graph.get_node_info(component.id).component_type == ComponentType.GENERATOR_SET
-                ],
-                TimeSeriesRate(
-                    values=[0.0] * self.variables_map.length,
-                    timesteps=self.variables_map.time_vector,
-                    unit=Unit.MEGA_WATT,
-                    rate_type=RateType.STREAM_DAY,
-                    regularity=regularity.values,
-                ),  # Initial value, handle no power output from components
+
+            # Ensure that only components at individual consumer level are included in aggregation of power,
+            # to secure that nothing is counted twice:
+            exclude_components = [
+                ComponentType.GENERATOR_SET,
+                ComponentType.ASSET,
+                ComponentType.INSTALLATION,
+                ComponentType.COMPRESSOR_SYSTEM,
+                ComponentType.CONSUMER_SYSTEM_V2,
+                ComponentType.PUMP_SYSTEM,
+                ComponentType.TRAIN_V2,
+            ]
+            power_electrical = self._compute_aggregated_power(
+                sub_components=sub_components,
+                regularity=regularity,
+                consumption_type=ConsumptionType.ELECTRICITY,
+                exclude_components=exclude_components,
             )
 
+            power_mechanical = self._compute_aggregated_power(
+                sub_components=sub_components,
+                regularity=regularity,
+                consumption_type=ConsumptionType.FUEL,
+                exclude_components=exclude_components,
+            )
+
+            power_sum = [
+                electrical + mechanical
+                for electrical, mechanical in zip(power_electrical.values, power_mechanical.values)
+            ]
+            power = copy.deepcopy(power_electrical)
+            power.values = power_sum
```

Comment:
add for TimeSeriesRate is implemented, I think. If not we could implement it. So that we can write 
`power = power_electrical + power_mechanical`

Reply:
Done